units/SDL3_gfxPrimitives.inc

```diff
@@ -0,0 +1,216 @@
+
+{* Polygon *}
+
+function polygonColor(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; color: cuint32): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_polygonColor' {$ENDIF} {$ENDIF};
+function polygonRGBA(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; r, g, b, a: cuint8): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_polygonRGBA' {$ENDIF} {$ENDIF};
+
+{* AA-Polygon *}
+
+function aapolygonColor(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; color: cuint32): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_aapolygonColor' {$ENDIF} {$ENDIF};
+function aapolygonRGBA(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; r, g, b, a: cuint8): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_aapolygonRGBA' {$ENDIF} {$ENDIF};
+
+{* Filled Polygon *}
+
+function filledPolygonColor(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; color: cuint32): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_filledPolygonColor' {$ENDIF} {$ENDIF};
+function filledPolygonRGBA(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; r, g, b, a: cuint8): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_filledPolygonRGBA' {$ENDIF} {$ENDIF};
+
+{* Textured Polygon *}
+
+function texturedPolygon(renderer: PSDL_Renderer; const vx, vy: pcint16; n: cint; texture: PSDL_Surface; texture_dx: cint; texture_dy: cint): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_texturedPolygon' {$ENDIF} {$ENDIF};
+
+{* Bezier *}
+
+function bezierColor(renderer: PSDL_Renderer; const vx, vy: pcint16; n, s: cint; color: cuint32): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_bezierColor' {$ENDIF} {$ENDIF};
+function bezierRGBA(renderer: PSDL_Renderer; const vx, vy: pcint16; n, s: cint; r, g, b, a: cuint8): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_bezierRGBA' {$ENDIF} {$ENDIF};
+
+{* Characters/Strings *}
+
+procedure gfxPrimitivesSetFont(const fontdata: Pointer; cw, ch: cuint32); cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_gfxPrimitivesSetFont' {$ENDIF} {$ENDIF};
+procedure gfxPrimitivesSetFontRotation(rotation: cuint32); cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_gfxPrimitivesSetFontRotation' {$ENDIF} {$ENDIF};
+function characterColor(renderer: PSDL_Renderer; x, y: cint16; c: cuint8; color: cuint32): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_characterColor' {$ENDIF} {$ENDIF};
+function characterRGBA(renderer: PSDL_Renderer; x, y: cint16; c, r, g, b, a: cuint8): Boolean; cdecl;
+  external GFX_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_characterRGBA' {$ENDIF} {$ENDIF};
+function stringColor(renderer: PSDL_Renderer; x, y: cint16; const s: pcuint8; color: cuint32): Boolean; cdecl;
```

Comment:
The argument `s` is a C char pointer, which we translate with `PAnsiChar` according to our guidelines.

[this applies for both string functions]